custom/enikshay/two_b_datamigration/management/commands/import_drtb_cases.py

```diff
@@ -833,26 +831,42 @@ def get_disease_site_properties(column_mapping, row):
     xlsx_value = column_mapping.get_value("site_of_disease", row)
     if not xlsx_value:
         return {}
-    if xlsx_value not in [
+
+    value = xlsx_value.replace('EP ', 'extra pulmonary ').\
+        lower().\
+        replace('extra pulmonary', 'extra_pulmonary').\
+        replace('lymph node', 'lymph_node').\
+        replace('pleural effusion', 'pleural_effusion')
+
+    if (not re.match("^extra_pulmonary \(other - .*$", value)
+        and value not in [
         "pulmonary",
         "extra_pulmonary",
+        "extra_pulmonary ",
         "extra_pulmonary (lymph_node)",
         "extra_pulmonary (spine)",
         "extra_pulmonary (brain)",
         "extra_pulmonary (pleural_effusion)",
         "extra_pulmonary (abdominal)",
         "extra_pulmonary (other)",
-    ]:
+    ]):
         raise FieldValidationFailure(xlsx_value, "site of disease")
-    classification = "extra_pulmonary" if "extra_pulmonary" in xlsx_value else "pulmonary"
-    match = re.match("\((.*)\)", xlsx_value)
+    classification = "extra_pulmonary" if "extra_pulmonary" in value else "pulmonary"
+    match = re.match("^.*\((.*)\)", value)
     if match:
         site = match.groups()[0]
+        if re.match("^other - .*$", site):
+            site_choice = site.replace('other - ', '')
+            site = 'other'
+        else:
+            site_choice = None
     else:
         site = None
+        site_choice = None
     return {
         "disease_classification": classification,
         "site_detail": site,
+        "site_choice": site_choice
     }
```

Comment:
Is it possible to avoid the complication instead?  Could the input spreadsheet be tailored to have three fields - classification, site_detail, and site_choice?
This looks fine if it works, just anticipating that there might be some difficulties running this on the real data.